sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -250,6 +252,8 @@ class Dataset[T] private[sql](
     val hasMoreData = takeResult.length > numRows
     val data = takeResult.take(numRows)
 
+    lazy val timeZone = TimeZone.getTimeZone(sparkSession.sessionState.conf.sessionLocalTimeZone)
```

Comment:
I think the preferred behavior would be that the result of `show()` should be changed by the session timezone.